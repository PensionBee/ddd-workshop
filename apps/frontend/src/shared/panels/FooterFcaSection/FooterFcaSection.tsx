import { GridContainer, GridItem, GridRow } from "@shared/components/Grid";
import { BodySmallL } from "@shared/components/Typography";

export const FooterFcaSection: React.FC = (props) => (
  <GridContainer
    data-rc="FooterFcaSection"
    pt={{ xs: 40, lg: 80 }}
    pb={{ xs: 40 }}
    {...props}
  >
    <GridRow pb={10}>
      <GridItem>
        <BodySmallL align="center" as="p">
          PensionBee is authorised and regulated by the Financial Conduct
          Authority.
        </BodySmallL>
        <BodySmallL align="center" as="p">
          With pensions, your capital is at risk.
        </BodySmallL>
      </GridItem>
    </GridRow>
  </GridContainer>
);
