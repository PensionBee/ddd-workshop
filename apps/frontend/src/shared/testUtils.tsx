import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import { ThemeProvider } from "styled-components";
import type { Breakpoints } from "./styling/breakpoints";
import { BREAKPOINTS } from "./styling/breakpoints";
import { theme } from "./styling/theme";
import { vi } from "vitest"

type ReactTestingLibraryProviderProps = {
  children: React.ReactNode;
};

// eslint-disable-next-line react-refresh/only-export-components
const ReactTestingLibraryProviders: React.FC<
  ReactTestingLibraryProviderProps
> = ({ children }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

// eslint-disable-next-line @typescript-eslint/no-explicit-any
const customRender = (ui: React.ReactElement, options?: any) => {
  return render(ui, { wrapper: ReactTestingLibraryProviders, ...options });
};

// re-export everything
// eslint-disable-next-line react-refresh/only-export-components
export * from "@testing-library/react";

// override render method
export { customRender as render };

export const sleep = (delay: number) => {
  return new Promise((resolve) => setTimeout(resolve, delay));
};

export const mockMediaMatches = (breakpoint: Breakpoints) => {
  Object.defineProperty(window, "matchMedia", {
    writable: true,
    value: (query: string) => {
      const breakpointValue = BREAKPOINTS[breakpoint];
      const queryValue = parseInt(query.replace(/^\D+/g, ""));
      const breakpointValues = [...Object.values(BREAKPOINTS)];
      if (!breakpointValues.includes(queryValue)) {
        throw new Error(
          `Invalid breakpoint value: ${queryValue}, this likely means there has been a change to the way that breakpoints are setup.`
        );
      }
      return {
        matches: queryValue <= breakpointValue,
        addEventListener: vi.fn(),
        removeEventListener: vi.fn(),
      };
    },
  });
};

export const routerRender = (ui: React.ReactNode, pathname = "/") => {
  return customRender(
    <MemoryRouter initialEntries={[{ pathname }]}>{ui}</MemoryRouter>
  );
};
