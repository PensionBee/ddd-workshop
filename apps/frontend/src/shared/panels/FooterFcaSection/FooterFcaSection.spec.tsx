import { render, vi } from "@shared/testUtils";
import { FooterFcaSection } from "./FooterFcaSection";

describe("FooterFcaSection", () => {
  const TEST_ID = "FooterFcaSection";
  it("should render", () => {
    const { getByTestId, asFragment } = render(
      <FooterFcaSection data-testid={TEST_ID} />
    );
    expect(getByTestId(TEST_ID)).toBeInTheDocument();
    expect(asFragment).toMatchSnapshot();
  });
});
