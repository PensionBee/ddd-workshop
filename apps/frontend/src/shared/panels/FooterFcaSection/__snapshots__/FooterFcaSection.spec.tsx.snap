// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterFcaSection should render 1`] = `[Function]`;
